kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -1739,13 +1717,13 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         }
 
         // Simply restore old LHS
-        (rA).swap(*mpOldAMatrix);
-        mpOldAMatrix = NULL;
+        rA.swap(mOldAMatrix);
```

Comment:
this should be a pointer swap (wrong even before)

Reply:
Ok

Reply:
I cannot do the swap with poiunter. I tried

~~~cpp
(&rA).swap(mpOldAMatrix);
~~~

And does not compile